app/src/main/java/com/example/pps_tudai/mvp/presenter/LoginPresenter.java

```diff
@@ -2,12 +2,13 @@
 
 import com.example.pps_tudai.mvp.model.LoginModel;
 import com.example.pps_tudai.mvp.view.LoginView;
-import static com.example.pps_tudai.utils.StringUtils.VALID_EMAIL_MODEL;
+import com.example.pps_tudai.utils.Validator;
 
 public class LoginPresenter {
 
     private final LoginModel loginModel;
     private final LoginView loginView;
+    private final Validator validator = new Validator();
```

Comment:
If validator only has a method, it may be a good idea to make it a static method within a static class.

Reply:
done!!!